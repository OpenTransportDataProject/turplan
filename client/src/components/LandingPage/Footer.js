import React from "react";
import styled from "styled-components";
import ntnu from "./images/ntnu_logo.png";
import sintef from "./images/sintef_logo.png";
import vegvesen from "./images/vegvesen_logo.png";

const FooterContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: rgb(58, 77, 109);
  padding-left: 10px;
  padding-right: 10px;
`;

const FooterText = styled.p`
  font-size: 1em;
  color: black;
  padding: 10px;
`;

const CustomerContainer = styled.div`justify-content: center;`;

const NTNUImg = styled.img`
  height: 50px;
  padding-right: 100px;
`;

const SINTEFImg = styled.img`
  height: 50px;
  padding: 10px;
`;

const VegvesenImg = styled.img`
  height: 50px;
  padding: 10px;
`;

export function Footer() {
  return (
    <FooterContainer>
      <NTNUImg src={ntnu} alt="ntnu" />
      <FooterText>
        {
          "TDT4290 Kundestyrt Prosjekt i samarbeid med SINTEF og Statens Vegvesen"
        }
      </FooterText>
      <CustomerContainer>
        <SINTEFImg src={sintef} alt="sintef" />
        <VegvesenImg src={vegvesen} alt="vegvesen" />{" "}
      </CustomerContainer>
    </FooterContainer>
  );
}

export default Footer;
