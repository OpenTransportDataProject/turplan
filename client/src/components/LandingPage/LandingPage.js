import React from "react";
import styled from "styled-components";
import Header from "./Header.js";
import Footer from "./Footer.js";

const Container = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

const ContentContainer = styled.div`
  flex: 8;
  display: flex;
  flex-direction: column;
  background-color: rgb(144, 173, 204);
`;

const SearchButton = styled.button`
  display: inline-block;
  padding: 5px;
  font-size: 1em;
  background-color: rgb(86, 115, 163);
  border: 2px solid rgb(58, 77, 109);
  border-radius: 2px;
  margin-left: 10px;
`;

const MapButton = styled.button`
  display: inline-block;
  padding: 20px;
  font-size: 1em;
  background-color: rgb(86, 115, 163);
  border: 2px solid rgb(58, 77, 109);
  border-radius: 2px;
`;

const Input = styled.input`
  width: 30%;
  padding: 0.5em;
  margin: 0.5em 0;
  border: 1px solid lightgray;
  border-radius: 3px;
  padding: 10px;
`;

const SearchContainer = styled.div`
  flex: 7;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 10px;
`;

const MapButtonContainer = styled.div`
  flex: 6;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding-top: 10px;
`;

export const LandingPage = () => {
  return (
    <Container>
      <Header />
      <ContentContainer>
        <SearchContainer>
          <Input placeholder="Søketekst" type="text" />
          <SearchButton
            onClick={() => console.log("Skal søke etter sted her...")}
          >
            Søk etter sted
          </SearchButton>
        </SearchContainer>
        <MapButtonContainer>
          <MapButton onClick={() => console.log("Skal gå til kart her...")}>
            Gå til kart
          </MapButton>
        </MapButtonContainer>
      </ContentContainer>
      <Footer />
    </Container>
  );
};

export default LandingPage;
