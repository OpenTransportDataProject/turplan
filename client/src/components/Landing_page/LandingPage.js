import React, { Component } from "react";
import Header from "../Header/Header.js";
import Footer from "../Footer/Footer.js";
import {
  ContentContainer,
  SearchButton,
  MapButton,
  Input,
  SearchContainer,
  PageContentContainer,
  PageContentHeader,
  PageBanner,
  Row,
  PageInfoContainer,
  Info,
  InfoHeader,
  InfoText,
  Image
} from "./LandingPageStyles";

//<PageBanner src={"images/tursti.jpg"}/>
class LandingPage extends Component {
  render() {
    return (
      <ContentContainer>
        <Header swapComponent={this.props.swapComponent.bind(this)}/>
        <PageContentContainer>
          <PageContentHeader>
            Denne siden er laget i samarbeid med Statens Vegvesen,
            Sintef Digital og studenter ved NTNU i forbindelse med emnet "Kundestyrt prosjekt" høsten 2017.
          </PageContentHeader>
          <PageInfoContainer>
            <Row>
              <Info>
                <InfoHeader>Finn parkeringsplasser</InfoHeader>
                <InfoText>I kartet kan du lett finne parkeringsplasser når du har aktivert parkering. Det hentes da inn informasjon fra to forskjellige kilder. Den første kilden, Open Street Map, er et api som inneholder parkeringsplasser lagt til av publikum. Disse kan mangele noe info om begrensinger, men det er lagt til store mengder data. De andre kilden er fra Statens Vegvesen, og disse inneholder mer info om begrensinger, men det er bare et lite subset av alle norges parkeringsplasser i dette registeret. Punktene lar seg ikke slå sammen, da de er registret på forskjellige måter.</InfoText>
              </Info>
              <Image imageUrl={"images/karteksempel.jpg"}/>
            </Row>
            <Row>
              <Image imageUrl={"images/lade.jpg"}/>
              <Info>
                <InfoHeader>Finn ladestasjon</InfoHeader>
                <InfoText>Ladestasjonene som er tilgjengelige i vårt kart er en kombinasjon av Open Street Map og Nobil Transnova's api. Sistnevnte samarbeider med elbilforeningen og har et ganske komplett bilde av alle norges ladestasjoner for elbil. Disse er utgangspunktet for vår applikasjon. Der det finnes ekstra fra Open Street Map har disse blitt lagt til, hvis ikke har vi valgt å vise Nobil's stasjoner. </InfoText>
              </Info>
            </Row>
            <Row>
              <Info>
                <InfoHeader>Finn turer</InfoHeader>
                <InfoText>Denne applikasjonen samler tur-informasjon fra Nasjonalturbase. Da denne var strukturert litt uhelig for vårt bruk, har vi lastet ned et subset av deres turer for å vise i vår applikasjon. </InfoText>
              </Info>
                <Image imageUrl={"images/tur.jpg"}/>
            </Row>
            <Row>
              <Image imageUrl={"images/kart.jpg"}/>
              <Info>
                <InfoHeader>Våre kart</InfoHeader>
                <InfoText>Vi benytter leaflet som basis for å vise kart-data og markere, med kartblad hentet ut fra statkart. </InfoText>
              </Info>
            </Row>
          </PageInfoContainer>
          <PageContentHeader>
            <h1>Api og datakilder</h1>
            <ul>
            <li>Nobil Transnova</li>
            <li>Open Street Map</li>
            <li>Statens vegvesen's parkeringsregister</li>
            <li>Google places</li>
            <li>Nasjonalturbase</li>
            <li>Leaflet</li>
            <li>Statkart/Kartverket/Norgeskart</li>

            </ul>
          </PageContentHeader>
          </PageContentContainer>
        <Footer />
      </ContentContainer>
    );
  }
}

export default LandingPage;
