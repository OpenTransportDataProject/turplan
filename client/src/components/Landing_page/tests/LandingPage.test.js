import React from 'react';
import LandingPage from '../LandingPage.js';
import renderer from 'react-test-renderer';

test('Renders correctly', () => {
  const mockFn = jest.fn();
  const component = renderer.create(
    <LandingPage swapComponent={mockFn}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
