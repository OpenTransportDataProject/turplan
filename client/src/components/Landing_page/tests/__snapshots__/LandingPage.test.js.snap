// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="sc-iwsKbI fwqsqj"
>
  <div
    className="sc-bdVaJa ebqGbM"
  >
    <h1
      className="sc-bwzfXH glACvo"
    >
      Turplan
    </h1>
    <div
      className="sc-htpNat cNrdsZ"
    >
      <h2>
        Her kan du finne turer og parkering
      </h2>
      <p>
        Nå kan du lett finne turer i ditt nærområde og planlegge turen dit. Her kan du hente ut parkeringsplasser og ladestasjoner til elbil ved noen få tastetrykk.
         
      </p>
      <button
        className="sc-bxivhb Dblxq"
        onClick={[Function]}
      >
        Gå til kart
      </button>
    </div>
  </div>
  <div
    className="sc-gqjmRU hSDwCS"
  >
    <p
      className="sc-VigVT ibqJrZ"
    >
      Denne siden er laget i samarbeid med Statens Vegvesen, Sintef Digital og studenter ved NTNU i forbindelse med emnet "Kundestyrt prosjekt" høsten 2017.
    </p>
    <div
      className="sc-jzJRlG eHwEw"
    >
      <div
        className="sc-fjdhpX dfJVIW"
      >
        <div
          className="sc-cSHVUG ndVwj"
        >
          <h3
            className="sc-kAzzGY cPHuHy"
          >
            Finn parkeringsplasser
          </h3>
          <p
            className="sc-chPdSV ipSNvI"
          >
            I kartet kan du lett finne parkeringsplasser når du har aktivert parkering. Det hentes da inn informasjon fra to forskjellige kilder. Den første kilden, Open Street Map, er et api som inneholder parkeringsplasser lagt til av publikum. Disse kan mangle noe info om begrensinger, men det er lagt til store mengder data. De andre kilden er fra Statens Vegvesen, og disse inneholder mer info om begrensinger, men det er bare et lite subset av alle norges parkeringsplasser i dette registeret. Punktene lar seg ikke slå sammen, da de er registret på forskjellige måter.
          </p>
        </div>
        <div
          className="sc-kgoBCf kmrWoU"
        />
      </div>
      <div
        className="sc-fjdhpX dfJVIW"
      >
        <div
          className="sc-kgoBCf ffHgPG"
        />
        <div
          className="sc-cSHVUG ndVwj"
        >
          <h3
            className="sc-kAzzGY cPHuHy"
          >
            Finn ladestasjon
          </h3>
          <p
            className="sc-chPdSV ipSNvI"
          >
            Ladestasjonene som er tilgjengelige i vårt kart er en kombinasjon av Open Street Map og Nobil Transnova's api. Sistnevnte samarbeider med elbilforeningen og har et ganske komplett bilde av alle norges ladestasjoner for elbil. Disse er utgangspunktet for vår applikasjon. Der det finnes ekstra fra Open Street Map har disse blitt lagt til, hvis ikke har vi valgt å vise Nobil's stasjoner.
             
          </p>
        </div>
      </div>
      <div
        className="sc-fjdhpX dfJVIW"
      >
        <div
          className="sc-cSHVUG ndVwj"
        >
          <h3
            className="sc-kAzzGY cPHuHy"
          >
            Finn turer
          </h3>
          <p
            className="sc-chPdSV ipSNvI"
          >
            Denne applikasjonen samler tur-informasjon fra Nasjonalturbase. Da denne var strukturert litt uhelig for vårt bruk, har vi lastet ned et subset av deres turer for å vise i vår applikasjon.
             
          </p>
        </div>
        <div
          className="sc-kgoBCf hgqDTh"
        />
      </div>
      <div
        className="sc-fjdhpX dfJVIW"
      >
        <div
          className="sc-kgoBCf cSCHWN"
        />
        <div
          className="sc-cSHVUG ndVwj"
        >
          <h3
            className="sc-kAzzGY cPHuHy"
          >
            Finn en rute
          </h3>
          <p
            className="sc-chPdSV ipSNvI"
          >
            For å finne en rute mellom punkter, benytter vi "Leaflet Routing Machine". I dette stadiet av applikasjonen, må ting gjøres i en bestemt rekkefølge for å fungere. Det første du trenger er en destinasjon. Du kan sette denne ved å klikke på markøren til enten en parkeringsplass eller ladestasjon, og så sette denne som ditt startpunkt. Videre klikker du på kartet dit du ønsker å reise fra. Da vil det automatisk genereres en rute fra ditt selvvalgte punkt på kartet til den valgte markøren, og du får opp en veibeskrivelse (på engelsk).
             
          </p>
        </div>
      </div>
      <div
        className="sc-fjdhpX dfJVIW"
      >
        <div
          className="sc-cSHVUG ndVwj"
        >
          <h3
            className="sc-kAzzGY cPHuHy"
          >
            Våre kart
          </h3>
          <p
            className="sc-chPdSV ipSNvI"
          >
            Vi benytter leaflet som basis for å vise kart-data og markere, med kartblad hentet ut fra statkart.
             
          </p>
        </div>
        <div
          className="sc-kgoBCf kBNJDJ"
        />
      </div>
    </div>
    <p
      className="sc-VigVT ibqJrZ"
    >
      <h1>
        Api og datakilder
      </h1>
      <ul
        className="sc-kGXeez bEiwLP"
      >
        <li>
          Nobil Transnova
        </li>
        <li>
          Open Street Map
        </li>
        <li>
          Parkeringsregisteret, Statens Vegvesen
        </li>
        <li>
          Google Places
        </li>
        <li>
          Nasjonal Turbase
        </li>
        <li>
          Leaflet routing machine
        </li>
        <li>
          Leaflet
        </li>
        <li>
          Statkart/Kartverket/Norgeskart
        </li>
      </ul>
    </p>
  </div>
  <div
    className="sc-ifAKCX dUPOVk"
  >
    <img
      alt="ntnu"
      className="sc-gzVnrw bpooqE"
      src="/images/ntnu_logo.png"
    />
    <p
      className="sc-EHOje jGhVpY"
    >
      TDT4290 Kundestyrt Prosjekt i samarbeid med SINTEF Digital og Statens Vegvesen
    </p>
    <div
      className="sc-bZQynM nyGYe"
    >
      <img
        alt="sintef"
        className="sc-htoDjs faPUxq"
        src="/images/sintef_logo.png"
      />
      <img
        alt="vegvesen"
        className="sc-dnqmqq bnXutj"
        src="/images/vegvesen_logo.png"
      />
       
    </div>
  </div>
</div>
`;
